utils/ProfileProvider.tsx

```diff
@@ -42,50 +39,67 @@ interface ProfileProviderProps {
   children: ReactNode;
 }
 export default function ProfileProvider({ children }: ProfileProviderProps) {
+  const { userId, loading: authLoading } = useAuth();
   const [profileData, setProfileData] = useState<Profile | null>(null);
   const [profileReady, setProfileReady] = useState<boolean>(false);
   const [hasPlot, setHasPlot] = useState<boolean | null>(null);
 
-  const loadProfile = useCallback(async (id?: UUID) => {
-    setProfileReady(false);
-
-    // If user isn't signed in, return null
-    if (!id && !placeholderUserId) {
+  const loadProfile = useCallback(async () => {
+    if (!userId) {
+      setProfileData(null);
       setProfileReady(true);
       return;
     }
-
-    const fetchedProfile = await fetchProfileById(placeholderUserId);
-
-    setProfileData(fetchedProfile);
-    setProfileReady(true);
-  }, []);
+    try {
+      setProfileReady(false);
+      const fetchedProfile = await fetchProfileById(userId);
+      setProfileData(fetchedProfile);
+    } catch (error) {
+      console.error('Error fetching profile:', error);
+    } finally {
+      setProfileReady(true);
+    }
+  }, [userId]);
 
   useEffect(() => {
-    loadProfile();
-  }, [loadProfile]);
+    if (!authLoading) {
```

Comment:
a possible edge case to consider is what if we want to call loadProfile (i.e. outside of this useEffect); perhaps we should handle the !authLaoding check inside loadProfile??

edit: the only place IJP used this was in /login, when deciding where to route the user. In the case of /login, we would already have access to authLoading, so we can leave it as is!